### DISCUSSION TITLE
How to set up middleware logger with log target as slog_redis?

### DISCUSSION THREAD

**Discussion by paulobarchi** [actix/actix-web#2870] [Q&A] [answered]
I'm introducing logging to an application (this is the open sourced version: https://github.com/bolcom/serenade; the MR I'm working on is from a private repo). I'm using `slog_redis` for logging what is outside `actix_web` (`slog`: structured logging; `slog_redis` for the RELK stack: https://github.com/bolcom/unFTP/tree/master/crates/redislog). I'm struggling to have the middleware logger to use the `slog_redis` so the logs are available in the RELK stack. 
I tried with `impl Redis for Logger {...}` and with something similar to [this discussion](https://stackoverflow.com/questions/32552593/is-it-possible-for-one-struct-to-extend-an-existing-struct-keeping-all-the-fiel), but could not make it work. Please, did anybody do anything similar? How did or would you approach it? Thanks!

**Comment by robjtede** [comment, marked as answer]
You would be best to copy-paste the Logger middleware and make the required changes. Actix web uses the `log` ecosystem extensively and does not provide a way to use `slog` instead.

**Reply by paulobarchi** [reply]
Got it. I was following that path, but since it uses much of the actix private modules and I would need to duplicate whole modules, I ended up trying other possible approaches. I'm going back to it then. In the meanwhile, I'd like to keep the discussion open in case anyone pops up with any other idea. Thanks!

**Reply by robjtede** [reply]
Might this be a better base? https://crates.io/crates/actix-slog

**Reply by paulobarchi** [reply]
Nice, thanks! I'll check that out.
